charmhelpers/contrib/network/ovs/__init__.py

```diff
@@ -303,3 +306,154 @@ def port_to_br(port):
         ).decode('UTF-8').strip()
     except subprocess.CalledProcessError:
         return None
+
+
+def _dict_to_vsctl(data, table, entity):
+    """
+    :param data: Additional data to attach to interface
+        The keys in the data dictionary map directly to column names in the
+        OpenvSwitch specified table as defined in DB-SCHEMA [0] referenced in
+        RFC 7047 [1]
+
+        There are some established conventions for keys in the external-ids
+        column of various tables, consult the OVS Integration Guide [2] for
+        more details.
+
+        NOTE(fnordahl): Technically the ``external-ids`` column is called
+        ``external_ids`` (with an underscore) and we rely on ``ovs-vsctl``'s
+        behaviour of transforming dashes to underscores for us [3] so we can
+        have a more pleasant data structure.
+
+        0: http://www.openvswitch.org/ovs-vswitchd.conf.db.5.pdf
+        1: https://tools.ietf.org/html/rfc7047
+        2: http://docs.openvswitch.org/en/latest/topics/integration/
+        3: https://github.com/openvswitch/ovs/blob/
+               20dac08fdcce4b7fda1d07add3b346aa9751cfbc/
+                   lib/db-ctl-base.c#L189-L215
+    :type data: Optional[Dict[str,Union[str,Dict[str,str]]]]
+    :param table: Name of table to operate on
+    :type table: str
+    :param entity: Name of entity to operate on
+    :type entity: str
+    :returns: '--' separated ``ovs-vsctl set`` commands
+    :rtype: Iterator[str]
```

Comment:
It doesn't appear to return a `str` .. should it be a `Tuple[str, str, str, str, str]` ?

Reply:
:+1: